src/main/java/com/instana/opentracing/InstanaSpanContext.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (c) Copyright IBM Corp. 2022
+ */
+package com.instana.opentracing;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.opentracing.SpanContext;
+import io.opentracing.propagation.TextMapExtract;
+
+class InstanaSpanContext implements SpanContext {
+
+  private final Map<String, String> items;
+
+  InstanaSpanContext(TextMapExtract carrier) {
+    this.items = new HashMap<String, String>();
+    for (Map.Entry<String, String> item : carrier) {
+      String key = item.getKey().toLowerCase();
```

Comment:
It's debatable if the canonical representation of Instana headers should be lower-case, upper-case, capitalized or whatelse.